coding-challenges.js

```diff
@@ -2,59 +2,162 @@
 
-
-
-
-
 var collections = ["purple", "blue", "green", "yellow", "pink"]
+
+// VERSION 1.0
+// found the solution using sort, but not quite sure how it actually shuffles
+// is it similar to sort's compare function – function(a,b) {a-b} for numbered strings?
+
+// Math.random generates a number between 0 and 0.9999
+// that number is then subtracted by 0.5, but how is that change the indices of the array?
+// if you know, please help explain
+const removeAndShuffleV1 = arr => {
+  arr.shift()
+  return arr.sort(() => Math.random() - 0.5);
+}
+
+console.log(removeAndShuffleV1(collections))
+
+//VERSION 2.0
+// shuffle using a for loop
+const removeAndShuffleV2 = arr => {
+  // removes the first element
+  arr.shift();
+  // loops through array, but starting from the last index
+  for (let i = arr.length - 1; i > 0; i--) {
+    // stores a random number between 0 and the variable i as declared above
+    const j = Math.floor(Math.random() * i);
+    // temp variable stores the value of the current element's value
+    const temp = arr[i];
+    // changes/switches the value of the current element
+    // to the value of the random number generated above
+    arr[i] = arr[j];
+    // changes the value of arr[j] to temp's value
+    arr[j] = temp;
+  }
+
+  return arr;
+}
+
+console.log(removeAndShuffleV2(collections))
 // Expected output example1 (can be a different order): ["yellow", "blue", "pink", "green"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
 
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
-
-
 
 
 
+// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 
+const sumCubed = arr => {
+  // (2 * 2 * 2) + (3 * 3 * 3) + (4 * 4 * 4) = 99
+  // 8 + 27 + 64 = 99
+  // maps thru and cubed each element
+  // used reduce to accumulate elements
+  // must set intial value to 0 or it will run incorrectly
+  return arr.reduce((acc, val) => acc + (val ** 3), 0);
+  // initial value = 0
+  // intiVal + (2**3) = 8
+  // 8 becomes the accumulator (acc)
+  // acc + (3**3) = 35
+  // 35 becomes the acc
+  // acc + (4**3) = 99
+}
+
+console.log(sumCubed(cubeAndSum1));
+console.log(sumCubed(cubeAndSum2));
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-
-
-
-
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
 
+const minMax = arr => {
+  return [Math.min(...arr), Math.max(...arr)];
+}
```

Comment:
excellent! just be prepared for an interviewer to ask you to write out the long form. i know you know how to do it, and this is very efficient code so good work :)